### PR TITLE
Add match_empty_levels as an option

### PR DESCRIPTION
**Pull request by gnought**

### CODE REVIEW

**Comment by gnought** [comment]
@mcollina would you mind bump a minor version to include `matchEmptyLevels` option when landed in master branch?

**Comment by gnought** [comment]
@mcollina done

**Comment by gnought** [comment]
If you don't mind, I could submit a PR to change another mqemitter repot to tap. Can I?

**Comment by mcollina** [comment]
I truly prefer tape for these abstract tests. The fact that it has not done a major in years is a good plus. In the same amount of time, tap has done ~10. Switching increase the coupling between all the various mqemitter repos and overall maintainance work.

(note I love tap and I use it for everything that is complex. These 1-file modules are not).

**Comment by gnought** [comment]
@mcollina done.

File: abstractTest.js [hunk 1]
```diff
@@ -1,37 +1,21 @@
...
-    var expected = {
+    const e = builder()
+    const expected = {
       topic: 'hello world',
       payload: { my: 'message' }
     }
 
     e.on('hello world', function (message, cb) {
-      t.equal(e.current, 1, 'number of current messages')
+      t.equal(e._current, 1, 'number of _current messages')
```
Comment (mcollina) [inline abstractTest.js]: why this change?

File: mqemitter.js [hunk 2]
```diff
@@ -1,61 +1,43 @@
...
   this._parallel = fastparallel({
     results: false,
     released: released
   })
 
-  this.concurrency = opts.concurrency
+  this.concurrency = opts.concurrency || 0
 
-  this.current = 0
+  this._current = 0
```
Comment (mcollina) [inline mqemitter.js]: please do not change this.

File: README.md [hunk 3]
```diff
@@ -52,96 +49,111 @@ emitter.emit(message, function () {
...
+__MQEmitter__ supports the use of wildcards: every topic is splitted according to `separator`.
 
-The wildcard character `+` matches exactly one word:
+The wildcard character `+` matches exactly _non-empty_ one word:
 
-```javascript
-var mq = require('mqemitter')
-  , emitter = mq()
+```js
+const mqt = require('mqemitter')
```
Comment (mcollina) [inline README.md]:
```suggestion
const mq = require('mqemitter')
```

File: mqemitter.js (hunk 2 above)
Comment (gnought) [reply mqemitter.js]: Is the `current` a public variable that used by others?

File: README.md (hunk 3 above)
Comment (gnought) [reply README.md]: Probably it is a typo.

**Comment by gnought** [comment]
I could split this PR into several, do you want me to do that?

File: mqemitter.js (hunk 2 above)
Comment (mcollina) [reply mqemitter.js]: it was identified as public. Changing it is a semver-major change. Moreover it’s useful.

**Comment by mcollina** [comment]
No need, I think we are mostly done.
